### ISSUE TITLE
`fish` won't build on FreeBSD

### ISSUE THREAD

**Issue by raichoo** [fish-shell/fish-shell#4491]
Hi,

I'm regularly building `fish` from source but it looks like something broke the build process for FreeBSD recently.

The `cmake` build fails with the following error message:

```
[  1%] Performing build step for 'muparser_project'
make[3]: make[3]: don't know how to make lib/libmuparser.a. Stop

make[3]: stopped in /usr/home/raichoo/Sources/fish-shell/build/muparser-build/obj
gmake[2]: *** [CMakeFiles/muparser_project.dir/build.make:111: muparser-build/src/muparser_project-stamp/muparser_project-build] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:716: CMakeFiles/muparser_project.dir/all] Error 2
gmake: *** [Makefile:130: all] Error 2
```
I've also tried the "good old" way of building it which yields the following:

```
$ autoreconf --no-recursive
gm4: m4_syscmd subprocess failed
cut: FISH-BUILD-VERSION-FILE: No such file or directory
gm4: m4_syscmd subprocess failed
cut: FISH-BUILD-VERSION-FILE: No such file or directory
gm4: m4_syscmd subprocess failed
cut: FISH-BUILD-VERSION-FILE: No such file or directory
$ ./configure                                                                           
configure: error: cannot find sources (fish) in . or ..
```

The last version I've built successfully is: `fish, version 2.6.0-688-g0a129475`

Kind regards,
raichoo

**Comment by ridiculousfish** [comment]
Thanks for filing this. What version of FreeBSD are you using?

**Comment by mqudsi** [comment]
Did you try `make distclean; make clean` first?

**Comment by raichoo** [comment]
@ridiculousfish I'm currently running 12-CURRENT. My build is from around 2 months ago.

**Comment by ridiculousfish** [comment]
FreeBSD12 isn't behaving at all for me with vagrant :( I don't think I'm going to be able to help debug this.

**Comment by mqudsi** [comment]
@raichoo you didn't respond to my question?

**Comment by coypoop** [comment]
the problem is that "make", as invoked by CMakeFiles/MuParser.cmake:18, isn't GNU make on the problem machines. with this changed to explicitly gmake I can get through.

**Comment by mqudsi** [comment]
@coypoop maybe, but that's not the problem in this thread. The legacy build should still work and the `./configure` script bailing doesn't have anything to do with cmake.

**Comment by mqudsi** [comment]
I think the problem is here: https://github.com/fish-shell/fish-shell/blob/master/build_tools/git_version_gen.sh#L1

861b55d7d changed that from `/bin/sh` to a hard dependency on `/bin/bash` 
Executing it directly without bash installed to `/bin/bash` results in

```
$ build_tools/git_version_gen.sh
-sh: build_tools/git_version_gen.sh: not found
```

Executing it via sh results in 
```
$ sh bd_tools/git_version_gen.sh
build_tools/git_version_gen.sh: ${BASH_SOURCE[...}: Bad substitution
usage: dirname string [...]
```

We can probably do better than `BASH_SOURCE`, especially since we are in control of the invocation. @zanchey ?

**Comment by mqudsi** [comment]
Fixed in e174b8a8008d95730f3f73e0c2bf99a9120a4588

Run `git clean -xdf` first, then `autoreconf --no-recursive` and `./configure` should then work.

**Comment by raichoo** [comment]
Thanks, works as expected now!
